### ISSUE TITLE
FR: differentiate between wifi connected and internet connectivity

### ISSUE THREAD

**Issue by Frenzie** [koreader/koreader#3160]
* KOReader version: 568e8f6
* Device: all

#### Issue
Being connected to the wifi would be sufficient to do stuff like calibre sync without having Internet connectivity.

See #3150

**Comment by rutapon** [comment]
Can it depend on feature?
When use Wiki and cloud, check internet connection.
And other (menu, ftp, calibre ) check only wifi connection.

But if we have to choose I agree with you, connecting to the wifi is sufficient.

**Comment by Frenzie** [comment]
That's what differentiate means. ;-)

**Comment by tigran123** [comment]
Yes, this is very desirable, because without this differentiation we cannot connect to an FTP server on a hotpoint AP such as an Android mobile. And without this ability Kobo Aura One is quite useless because of its tiny internal storage (non-upgradeable) and the lack of external micro sd support.

**Comment by Frenzie** [comment]
Copying this here from https://github.com/koreader/koreader/issues/3273#issuecomment-333209565 for clarity.

The interim command for gateway ping I came up with looks like this:
```
ping $(ip route | grep '^default' | awk '/eth0/ {print $3}')
```

The structure I suggested for the code is this:
```lua
-- wifi only, pings gateway
NetworkMgr:isConnected()

-- internet connectivity, presumably depends on isConnected and also pings MS server
NetworkMgr:isOnline()
```

**Comment by tigran123** [comment]
The command `ping` by default will send infinitely many packets. You should specify `-c1` option to tell it to emit only a single packet.

**Comment by Frenzie** [comment]
Of course. My concern was the other part, but updated for clarity. :-)

**Comment by Frenzie** [comment]
Implemented in #3593.
